addons/edi/models/models.py

```diff
@@ -39,13 +39,40 @@ def groupby(self, key, sort=True):
     a non-arbitrary ordering is required, then use :meth:`~.sorted` to
     sort the recordset first, then pass to :meth:`~.groupby` with
     ``sort=False``.
+
+    Any recordsets within a tuple in ``key`` will be replaced with a list of IDs from each
+    recordset.
     """
+    def tuple_contains_recordset(tup):
+        """Returns True if the supplied tuple contains a recordset, otherwise False"""
+        for item in tup:
+            if isinstance(item, models.BaseModel):
+                return True
+        return False
+
+    def get_ids_from_recordset_in_tuple(tup):
+        """
+        Replaces each Recordset found within the supplied tuple
+        with a list of ids from that recordset
+        """
+        temp_key = []
+        for item in tup:
+            next_key = item
+            if isinstance(next_key, models.BaseModel):
```

Comment:
Note: If we do keep this approach, might want to do a check for tuple within a tuple, and check that for recordsets also... :)